### PR TITLE
Stale queue selection

### LINKED ISSUE #1679: ignore stale queues during selection
related to #1676, #1661, #1640, #1677 

now we're touching update_at just ignore any queues that are a more than an 30 mins old

### PR DESCRIPTION
**Pull request by camallen**
closes #1679, #1661, #1640, #1677 - request fresh selection data when queues are stale (older than 30 mins). 

Refactored the selection code which is a chunk of this PR to it's own strategy selector.

### CODE REVIEW

File: app/models/subject_queue.rb [hunk 1]
```diff
@@ -71,6 +72,12 @@ def update_ids(sms_ids)
     update_attribute(:set_member_subject_ids, capped_sms_ids)
   end
 
+  def stale?
+    return false unless updated_at
+    last_updated_mins_ago = (Time.zone.now - updated_at).to_i / 60
```
Comment (marten) [inline app/models/subject_queue.rb]: Would `updated_at < STALE_MINS.minutes.ago` work here?

**Comment by camallen** [comment]
updated

File: app/workers/enqueue_subject_queue_worker.rb [hunk 2]
```diff
@@ -13,64 +13,18 @@ class EnqueueSubjectQueueWorker
...
-    @selection_strategy = strategy(strategy_override)
-
-    sms_ids = strategy_sms_ids
-    unless sms_ids.empty?
-      user_seens = user ? UserSeenSubject.find_by(user: user, workflow: workflow) : nil
-      unseen_ids = Subjects::SeenRemover.new(user_seens, sms_ids).unseen_ids
-      queue.enqueue_update(unseen_ids)
-    end
+    queue = SubjectQueue.find(queue_id)
+    selector = Subjects::StrategySelector.new(
```
Comment (marten) [inline app/workers/enqueue_subject_queue_worker.rb]: I was slightly confused for a second that this doesn't select a **strategy**. Not sure what a better name would be, maybe `SelectsUsingStrategy` or something terrible like that?

Comment (marten) [inline app/workers/enqueue_subject_queue_worker.rb]: `StrategyBasedSelector`?

Love the extraction itself though.

Comment (camallen) [inline app/workers/enqueue_subject_queue_worker.rb]: I didn't like the name either...will come up with something enterprisey for you!

Comment (camallen) [inline app/workers/enqueue_subject_queue_worker.rb]: How about `StategySelection`?

Comment (camallen) [inline app/workers/enqueue_subject_queue_worker.rb]: How about `StrategySelection`?

Comment (marten) [inline app/workers/enqueue_subject_queue_worker.rb]: Sounds good. Matches PostgresqlSelection and stuff. Like it!

**Comment by marten** [comment]
LGTM

**Pull request merged**